pkgs/applications/virtualization/virt-viewer/default.nix

```diff
@@ -22,8 +22,9 @@ stdenv.mkDerivation rec {
   buildInputs = [
     pkgconfig intltool glib libxml2 gtk3 gtkvnc gmp libgcrypt gnupg cyrus_sasl
     shared_mime_info libvirt libcap_ng yajl gsettings_desktop_schemas makeWrapper
-    xen numactl
-  ] ++ optionals spiceSupport [ spice_gtk spice_protocol libcap gdbm ];
+    numactl
+  ] ++ optionals spiceSupport [ spice_gtk spice_protocol libcap gdbm
+  ] ++ optional stdenv.is64bit xen;
```

Comment:
That is why ```doesWorkOn``` would be better than spreading ```stdenv.system == "x86_64-linux"``` across nixpkgs.

```optional (stdenv.lib.lists.elem stdenv.system xen.meta.platforms) xen``` does not work for me:
```Package ‘xen-4.5.5’ in ‘/etc/nixos/nixpkgs/pkgs/applications/virtualization/xen/generic.nix:184’ is not supported on ‘i686-linux’, refusing to evaluate.```